### DISCUSSION TITLE
Temporarily disable devnet from CI/CD pipeline?

### DISCUSSION THREAD

**Discussion by ivpavici** [seanjameshan/starknet.js#327] [General]
Hi!

Context: 
- after 0.10 starknet release, [devnet](https://github.com/Shard-Labs/starknet-devnet/tree/master/starknet_devnet) had to go through a lot of changes, but relevant to starknet.js:

    - OZ accounts are not yet modified to support 0.10 - https://github.com/OpenZeppelin/cairo-contracts/pull/449
    - `estimate_fee` is not yet working as it should according to 0.10 changes (needs to support transaction types “DECLARE”, “INVOKE_FUNCTION”) - **note**: there is a possibility that this problem is first related to the straknet CLI, and then devnet... Fabijan and Ariel discussed about this couple of days ago ... https://github.com/Shard-Labs/starknet-devnet/issues/266

- starknet.js relies upon the devnet as the first level of tests in our CI/CD pipeline -> and now these tests are [failing](https://github.com/0xs34n/starknet.js/runs/8248623637?check_suite_focus=true) for https://github.com/0xs34n/starknet.js/pull/318

2 options:

1.  Wait for devnet to adapt 

    - regarding AccountContract - either by waiting for OZ, switch to Argenst accounts, or some 3rd option

2. Disable the devnet temporarily from the pipeline until the issues with devnet are resolved

    - in this case, all tests would still get run on testnet sequencer / rpc node

Please share thoughts :)

**Comment by janek26** [comment]
In my mind these changes should not take long for devnet to adopt. 
Switching to Argent contracts makes a lot of sense to me, as they behave the same and are always much faster updated to a new starknet version. Not just for 0.10

If you guys all think you should disable devnet and I‘m the only one strict against it, feel free to move forward please

**Comment by ivpavici** [comment]
https://github.com/Shard-Labs/starknet-devnet/releases/tag/v0.3.1

new devnet released, seems that issues are now resolved, thanks @dhruvkelawala !

**Comment by tabaktoni** [comment]
Resolved
